File: staj/stajApi/Models/ChatMessage.cs
using System.ComponentModel.DataAnnotations;

public class ChatMessage
{

    public int Id { get; set; }

    public int? SenderId { get; set; }           // User.Id veya Customer.Id
    public string? SenderRole { get; set; }      // "User" veya "Customer"

    public int? ReceiverId { get; set; }
    public string? ReceiverRole { get; set; }    // "User" veya "Customer"

    public string? SenderName { get; set; }
    public string? ReceiverName { get; set; }

    [Required(ErrorMessage = "Mesaj boş olamaz")]
    [MinLength(1, ErrorMessage = "Mesaj en az 1 karakter olmalıdır")]
    public string? Message { get; set; }
    public DateTime? Timestamp { get; set; } = DateTime.Now;
}

File: MobilCustomer/lib/Models/ChatMessage.dart
class ChatMessage {
  final int? senderId;
  final String? senderRole;
  final int? receiverId;
  final String? receiverRole;
  final String? message;
  final String? senderName;
  final String? receiverName;
  final DateTime? timestamp;

  ChatMessage({
    this.senderId,
    this.senderRole,
    this.receiverId,
    this.receiverRole,
    this.message,
    this.senderName,
    this.receiverName,
    this.timestamp,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      senderId: json['senderId'],
      senderRole: json['senderRole'],
      receiverId: json['receiverId'],
      receiverRole: json['receiverRole'],
      message: json['message'],
      senderName: json['senderName'],
      receiverName: json['receiverName'],
      timestamp: DateTime.tryParse(json['timestamp'] ?? ""),
    );
  }
}
